OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -179,13 +179,8 @@ public override bool Tick(Actor self)
 
 			// Determine where to search from and how far to search:
 			var procLoc = GetSearchFromProcLocation(self);
-			var searchFromLoc = lastHarvestedCell ?? procLoc;
+			var searchFromLoc = lastHarvestedCell ?? procLoc ?? self.Location;
 			var searchRadius = lastHarvestedCell.HasValue ? harvInfo.SearchFromHarvesterRadius : harvInfo.SearchFromProcRadius;
-			if (!searchFromLoc.HasValue)
-			{
-				searchFromLoc = self.Location;
-				searchRadius = harvInfo.SearchFromHarvesterRadius;
```

Comment:
Won't this mean we use `SearchFromProcRadius` in case we use `self.Location`?

Reply:
Yes, that is the idea. Without this, the range for the initial search is too short, causing harvesters in the shell maps to stall.

Apart from the initial search, `Self.Location` is only really used for the center of the search as a last resort.

Reply:
Sigh, guess I need to finally revisit #18198 to get rid of this mess. (Or at least reduce it.)

Reply:
I can take that over if you like. After all, I did the the big harvester refactor last time and some issues where deliberately left hanging then. I would be happy to finish what I started.

Reply:
If you want to go for it. I'm just no longer sure if it should search from the refinery first. Probably it still should, but return to the refinery to unload first before doing more scanning in case no ore is in the refinery radius.

Reply:
I already have some ideas on how to deal with this. I will probably split things in multiple PRs and take on some related dangling issues with harvester behaviour while I'm at it, so that we can close this topic once and for all. I would like to get this PR merged first though before I go further, because it is all closely related and it makes more sense to handle this in a linear fashion.

Reply:
Do we want the has dependencies labels on this PR then?